src/v/net/connection_rate.h

```diff
@@ -29,8 +31,105 @@ using namespace std::chrono_literals;
 namespace net {
 
 struct connection_rate_info {
-    int64_t max_connection_rate;
+    std::optional<int64_t> max_connection_rate;
     std::vector<ss::sstring> overrides;
 };
 
-} // namespace net
\ No newline at end of file
+enum class should_spawn_fiber_on_update { yes, no };
```

Comment:
nit: this could be a `ss::bool_class` for uniformity with other places we create yes/no types.